names/names.py

```diff
@@ -10,15 +10,46 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []
+
+# start_time = time.time()
+
+# duplicates = []
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+# end_time = time.time()
+# print (f"{len(duplicates)} duplicates:\n\n{', '.join(duplicates)}\n\n")
+# print (f"runtime: {end_time - start_time} seconds")
+# time complexity of O(n ^ 2)? Quadratic 
+print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`")
+
+# #OPTION 1:  cant use set though blah
+# start_time = time.time()
+# duplicates = set(names_1) & set(names_2)
+# end_time = time.time()
+# print (f"{len(duplicates)} duplicates:\n\n{', '.join(duplicates)}\n\n")
+# print (f"runtime: {end_time - start_time} seconds")
+print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`")
+
+# OPTION 2:
+start_time2 = time.time()
+end_time2 = time.time()
+
+bst = BinarySearchTree('names')
```

Comment:
Great solution here! Yeah, just let the regression make that BST for you hahahha!